library/ns1_team.py

```diff
@@ -0,0 +1,546 @@
+                        options=dict(
+                            manage_global_2fa=dict(type="bool", default=False),
+                        ),
+                    ),
+                    dns=dict(
+                        required=False,
+                        type="dict",
+                        default=None,
+                        options=dict(
+                            zones_allow=dict(type="list", default=False),
+                            manage_zones=dict(type="bool", default=False),
+                            zones_deny=dict(type="list", default=False),
+                            view_zones=dict(type="bool", default=False),
+                            zones_allow_by_default=dict(type="bool", default=False),
+                        ),
+                    ),
+                ),
+            ),
+            state=dict(
+                required=False,
+                type="str",
+                default="present",
+                choices=["present", "absent"],
+            ),
+        )
+
+        NS1ModuleBase.__init__(
+            self,
+            self.module_arg_spec,
+            supports_check_mode=True,
+        )
+
+    @Decorators.skip_in_check_mode
+    def update(self, team_id, built_changes):
+        """Updates a team with permissions from task
+
+        :param team_id: Zone object of existing zone returned by NS1
+        :type team_id: str
+        :param built_changes: Dict of permissions to be applied to a new
+        team.
+        :type built_changes: dict
+        :return: The updated zone object returned by NS1
+        :rtype: dict
+        """
+        team_update = self.ns1.team()
+        return team_update.update(team_id, **built_changes)
+
+    @Decorators.skip_in_check_mode
+    def create(self, built_changes):
+        """Creates a team with the given permissions.
+
+        :param built_changes: Dict of permissions to be applied to a new
+        team.
+        :type built_changes: dict
+        :return: [description]
+        :rtype: [type]
+        """
+        team_create = self.ns1.team()
+        return team_create.create(**built_changes)
+
+    @Decorators.skip_in_check_mode
+    def delete(self, team_id):
+        """Deletes a team.
+
+        :param team_id: Id of an existing team.
+        :type team_id: str
+        """
+        team_delete = self.ns1.team()
+        team_delete.delete(team_id)
+
+    def build_permissions(self):
+        """Builds a complete set of permissions based on defaults with values
+        updated by task parameters.
+
+        :return: A complete set of permissions.
+        parameters.
+        :rtype: dict
+        """
+        default_permissions = dict(permissions=_default_perms)
+        built_permissions = copy.deepcopy(default_permissions)
+        for key in default_permissions["permissions"]:
+            if self.module.params["permissions"] is None:
+                built_permissions = default_permissions
```

Comment:
No, this was an oversight. We actually dont need to set this variable at all since `built_permissions` is a copy of `default_permissions` to start out with.
The function has been re-written as:
```
    def build_permissions(self):
        """Builds a complete set of permissions based on defaults with values
        updated by task parameters.

        :return: A complete set of permissions.
        :rtype: dict
        """
        default_permissions = dict(permissions=_default_perms)
        built_permissions = copy.deepcopy(default_permissions)
        for key in default_permissions["permissions"]:
            if "permissions" in self.module.params:
                if isinstance(self.module.params["permissions"], dict):
                    if key in self.module.params["permissions"]:
                        if isinstance(self.module.params["permissions"][key], dict):
                            for key_2, value_2 in self.module.params["permissions"][
                                key
                            ].items():
                                built_permissions["permissions"][key][key_2] = value_2
        return built_permissions
```